### DISCUSSION TITLE
Raspberrymatic erkennt HMIP-RFUSB-TK nicht

### DISCUSSION THREAD

**Discussion by 73Anderl** [OpenCCU/OpenCCU#2256] [🇩🇪 Fragen&Antworten] [answered]
Hallo,

ich habe auf einer RaspberryPi 3b Rasberrymatic auf die SD Karte gespielt.
Das startet soweit auch alles, jedoch wird der HMIP-RFUSB-TK nicht erkannt (nichts im Posteingang)
Bei "Carrier Sense" steht n/a.
Habe ich was vergessen zu installieren?
Ich bin davon ausgegangen der USB Stick wird beim ersten booten automatisch erkannt.

**Comment by jens-maus** [comment]
Zeig den Inhalt der `/var/log/boot.log` Datei bzw. ein Screenshot der Hilfe-Seite der WebUI wo auch technische Detailinfos im rechten Bereich zu finden sind.

Und das der CarrierSense mit "n/a" ausgegeben wir ist beim `HmIP-RFUSB-TK` (der mit "TK" für Telekom) normal, da dieser mit einer alten Firmware daherkommt der neben dem fehlenden Support für das Ausgeben des CarrierSense auch kein Support für das alte nicht-HmIP Protokoll (BidCos-RF) sowie kein Support für das Anbinden eines `HmIP-HAP` oder `HmIPW-DRAP` mit sich bringt. Ergo ist der `HmIP-RFUSB-TK` aus vielerlei Hinsicht keine gute Wahl und man sollte besser die nicht-TK Variante einsetzen die man direkt von elv.de beziehen kann.

**Comment by 73Anderl** [comment]
Hallo,

wollte nicht löten, und den HmIP-RFUSB hatte eine unbekannte Lieferzeit....
Ich habe noch nicht versucht einen Homematik Aktor oder Schalter anzulernen, weil ich dachte ich habe kein funktionierendes Funkmodul

![image](https://user-images.githubusercontent.com/126382046/224373864-2321d14f-c932-44a4-984a-a579875348a1.png)

**Reply by MichaelN0815** [reply]
Das Funkmodul itself landet nie im Posteingang.

Das wäre ja so, als würdest du bei Amazon einen Briefkasten bestellen und der Briefträger soll ihn in den Briefkasten zustellen... Henne, Ei...

**Reply by jens-maus** [reply]
Passt doch alles soweit ich sehen kann. Wenn der Anlernmodus losrennt passt doch alles (bis auf die Einschränkungen die ich benannt habe).

**Reply by Baxxy13** [reply]
> Das Funkmodul itself landet nie im Posteingang.

Der HmIP-RFUSB schon. Den habe ich schon x-mal aus dem Posteingang geholt.

**Reply by MichaelN0815** [reply]
Oh. Das wusste ich nicht

**Reply by 73Anderl** [reply]
oki doki, dann werde ich mal versuchen meinen Jalousien Aktor anzulernen

**Reply by Basti67** [reply]
Das der HmIP-RFUSB-TK nicht die beste Wahl ist habe ich verstanden. Da ich aber nur HmIP Geräte habe sollte dieser eigentlich für mich reichen. Dazu kommt das ich leider nicht mehr löten kann und ich daher den Bausatz meiden möchte.
Ich habe mir einen frischen HA auf Raspberry Pi 4 aufgesetzt und Raspberrymatic installiert. Auch bei mir erscheint leider der Stick nicht im Posteingang und mein Info Bereich ist ähnlich wie bei 73Anderl. Wie kann ich dies Problem mit diesem Stick beheben?

CCU Hardware Info
Serial Number:	1D89926642
Hardware Model:	n/a (oci)
CPU / Memory:	Cortex-A72 (4), 7.6G
Storage:	223.6G SA400S37240G /dev/sda
Real-Time-Clock:	n/a
CCU Software Info
Product:	raspmatic_oci_arm64 (3.67.10.20230225)
ReGaHss:	R1.00.0388.0233
Engines:	Tcl (8.6.12), Java (1.8.0_362), NodeJS (v16.18.1)
Status:	SD(1) NTP(0) Link(1) Internet(1) IP(1)
Operating System Info
OS Type/Kernel:	Buildroot 2022.11.1 (Linux 5.15.84-v8 aarch64)
Uptime:	0 d, 0 h 45 min
Load Average:	0.49 0.40 0.40
System Temperature:	39.4 °C
Memory / Swap Utilization:	14.3 % / 0.0 %
NTP Offset (Server):	n/a (n/a)
rootfs / userfs Free Space:	206.2 GB [94.0%] / 206.2 GB [94.0%]
homematicIP-RF (HmIP) Info
RF-Module/Firmware:	HMIP-RFUSB-TK (4.2.14)
Device-Node:	/dev/raw-uart (eQ-3 HmIP-RFUSB@usb-0000:01:00.0-1.4)
Address:	0xB0E39C (0xB7CA32)
SGTIN:	3014F5AC9400041D89926642
Serial:	1D89926642
DC / CS:	0 % / -1 %
HomeMatic-RF (HmRF) Info
RF-Module/Firmware:	n/a (n/a)
Device-Node:	n/a (n/a)
Address:	n/a (n/a)
Serial:	n/a
DC / CS:	% / %

**Reply by Baxxy13** [reply]
Das ist bei der -TK Version so und völlig normal.
Der landet niemals im Posteingang und das es keine Info zu "HomeMatic-RF" gibt liegt einfach daran das der -TK das Homematik-Funk-Protokoll nicht unterstützt.

Was ist denn das eigentliche Problem?
Lassen sich keine HmIP Aktoren / Sensoren anlernen?

**Comment by friedpa** [comment]
Gerne löte ich Ihnen einen Bausatz zusammen. Wenn Sie interesse haben, schicken Sie mir ein PM.

**Comment by hardl1** [comment]
Hallo, 
seit etwa einem Jahr läuft RaspberryMatic in einer VS auf einem QNAP NAS.
Seit 2 Tagen werden  die Geräte nicht mehr gefunden.

In der VS wird der RFUSB angezeigt:
<img width="664" alt="Bildschirmfoto 2024-02-29 um 10 38 03" src="https://github.com/jens-maus/RaspberryMatic/assets/38688960/08966cef-d007-4e12-93f5-dfa0279df6f9">
<img width="495" alt="Bildschirmfoto 2024-02-29 um 10 40 14" src="https://github.com/jens-maus/RaspberryMatic/assets/38688960/876bd0eb-1f7b-4876-891f-964364023bf0">

In der Raspberrymatic aber nicht:
<img width="441" alt="Bildschirmfoto 2024-02-29 um 10 24 35" src="https://github.com/jens-maus/RaspberryMatic/assets/38688960/09db147d-5944-4cf8-8bfa-0f1d1aab197d">

Hat jemand eine Idee?

**Comment by hardl1** [comment]
Hat sich erledigt.
Ich habe eine neue VM erstellt, das brachte aber auch nichts.
Dann wieder auf die ursprüngliche VM zurück.
Nach mehrmaligem Herunterfahren der QNAP oder der RaspberryMatic und Hinzufügen des RF-USB Stick ging es plötzlich wieder.
Ich habe 2Tage lang bestimmt 20 Versuche gestartet, aber was jetzt half, weiß ich leider nicht.

**Reply by jens-maus** [reply, marked as answer]
An RaspberryMatic selbst wird es nicht liegen. Solche Probleme sind meistens vom Host-System der Virtualisierung ausgehend. D.h. der Host (QNAP) wird einfach den USB Stick selbst nicht richtig erkannt haben oder konnte ihn aus irgendwelchen Gründen nicht an die VM durchleiten. Da hilft nur das QNAP selbst zu debuggen oder eben mal ein Update von QNAP zu machen oder sicherzustellen das QNAP sich selbst nicht den Stick reserviert oder so. Das nächste mal einfach mal unter RaspberryMatic auf der konsole als Befehl `lsusb` eingeben um zu schauen welche USB Geräte faktsich durchgeleitet werden. Und wenn da der `HmIP-RFUSB` nicht dabei ist ist eben QNAP selbst schuld und dann brauch es dort einen Reboot oder sonstigen Eingriff.

**Comment by hardl1** [comment]
Hallo jens,
vielen Dank für Deine ausführliche Antwort. 
Ich werde das beim nächsten Mal prüfen und gegebenenfalls wieder berichten.

**Comment by hardl1** [comment]
Hallo Jens,

leider muss ich mich nochmals melden.
Gestern hatte ich einen Stromausfall, danach waren die HmIP Geräte nicht mehr erreichbar.

Ich konnte nach einigen Versuchen den RFUSB wieder einbinden, er wird unter USB1 angezeigt.
RaspberryMatic startet, es sind unter  Einstellungen alle Geräte vorhanden. 
Auf der Startseite dreht "LOADING" ewig und bei Bedienung sind nur 2 CuXD-Geräte vorhanden.

Ich wollte in der RaspberryMatic Konsole lsusb eingeben, kann aber in der Konsole nichts eingeben, auch mit "ALT-F2" tut sich nichts.
Lautet der Befehl beim Macbook anders?

<img width="585" alt="Bildschirmfoto 2024-03-08 um 08 06 42" src="https://github.com/jens-maus/RaspberryMatic/assets/38688960/c076f905-8ecf-442c-9a44-daeb261fb691">

**Comment by hardl1** [comment]
Gibt es noch andere Ideen?

Vielleicht ist ja mein HmIP-RFUSB Stick defekt, aber man kann derzeit nirgends einen kaufen um das zu testen.

Ich habe jetzt mehrmals die QNAP und Raspberrymatic neu gestartet. Der Stick wird immer erkannt, aber scheint keine Verbindung zu haben.

Auch die Lötstellen habe ich sicherheitshalber nachgelötet.

**Reply by jens-maus** [reply]
Dann probier den `HmIP-RFUSB` doch mal ausserhalb von QNAP z.B. mit einem Pi auf dem dann RaspberryMatic läuft..

**Reply by Baxxy13** [reply]
Genau, mal ein einfaches Setup (RM auf Pi2 / Pi3) aufsetzen und damit testen.
Vielleicht können eine paar Infos aus meinem Thread zum "Problem"-Stick gezogen werden.
[Link](https://homematic-forum.de/forum/viewtopic.php?f=31&t=79828&hilit=problemkind#p776803)

**Comment by hardl1** [comment]
Also, ich habe Raspberrymatic auf einem Raspi4 installiert und eine HmIP Steckdose einrichten können.

Zurück auf der VM wird der Stick zwar erkannt, unter USB1 in der Hilfe sieht es so aus:
<img width="442" alt="Bildschirmfoto 2024-03-10 um 10 35 47" src="https://github.com/jens-maus/RaspberryMatic/assets/38688960/4a27de26-83b3-419b-a7e8-8218e2a83525">

Dann habe ich eine neue VM erstellt, ging auch nicht.

Wieder zurück auf die ursprüngliche VM und alles funktioniert einwandfrei.

Um die neue VM nochmals zu testen bin ich zurück, wieder kein Erfolg.

Jetzt zurück auf die ursprüngliche VM, läuft auch nicht mehr.

Ich gehe davon aus, dass es an der USB-Verbindung der QNAP liegt.

**Comment by friedpa** [comment]
Das Problem liegt daran, dass sich die QNAP intern die USB Schnittstelle reserviert.

**Comment by hardl1** [comment]
Was kann man dagegen tun?
Es ging ja  über 1 Jahr.

**Reply by jens-maus** [reply]
Das liest sich für mich alles danach das man QNAP selbst einfach bei USB settings ggf. `Disallow USB Devices` machen sollte und dann kann man ggf. in der Virtualisierung von QNAP dann das USB Gerät der RaspberryMatic VM zuweisen. Siehe:

https://docs.qnap.com/operating-system/qts/5.0.x/en-us/configuring-usb-settings-2176DEDB.html

Wenn das geht muss man aber wohl damit leben das man dann innerhalb von QNAP selbst keine USB Geräte nutzen kann. Vielleicht gibt es ja aber auch irgendein "Trick" das man dem Linux-System von QNAP sagen kann das es sich den `HmIP-RFUSB` nicht schnappen soll. Müsste man mal tiefer recherchieren...

**Comment by hardl1** [comment]
Leider hat auch das Abschalten der USB-Geräte in QNAP keine Änderung gebracht. 
Ich kann den USB-Stick wählen, aber der RF-USB findet nichts.
Es ist wahrscheinlich besser Raspberrymatic auf ein anderes System zu übertragen.

**Reply by jens-maus** [reply]
> aber der RF-USB findet nichts. 

Was heisst denn hier bitte "findet nichts"? D.h. der RFUSB wird beim Hochfahren erkannt und ist auch unter der Hilfeansicht in der WebUI zu sehen und der Anlernmodus startet und zählt runter aber nichts landet im Posteingang? Wenn das so ist hast du ggf. ein anderes Problem (Carrier Sense). Nur wenn der RFUSB schon gar nicht beim Hochfahren erkannt wird und nicht angebunden wird an RM kann es das hier beschrieben Problem sein. Und wenn du auch einen `HmIP-RFUSB-TK` (mit TK hinten) hast, dann spricht alles dafür das du einfach zuviel Störstrahlung in der nähe des QNAP hast und deshalb das HmIP-RFUSB-TK keine Geräte finden kann (weil es Listen-before-talk macht). Und leider hat der HmIP-RFUSB-**TK** keine CarrierSense Anzeige/Ausgabe und kann dir daher nicht sagen das die Störstrahlung zu hoch ist..

**Comment by hardl1** [comment]
Es ist kein "-TK".
Der RF-USB wird in der VM erkannt.
In der Hilfe erscheint er, aber ohne Daten, wie im Bildschirmfoto von gestern.

Meine Geräte sind in den Einstellungen wie immer vorhanden.

In Status und Bedienung -> Geräte sind nur die beiden CUxD-Geräte vorhanden.

In der Startseite dreht Loading ewig und CarrierSense ist nicht zu sehen.

<img width="1584" alt="Bildschirmfoto 2024-03-11 um 08 56 07" src="https://github.com/jens-maus/RaspberryMatic/assets/38688960/53b59311-9cd7-4485-8f30-0af2abda4b42">

**Reply by jens-maus** [reply]
> Der RF-USB wird in der VM erkannt. In der Hilfe erscheint er, aber ohne Daten, wie im Bildschirmfoto von gestern.

Diese Aussagen beissen sich aber. Wo wird der "in der VM erkannt"? Du meinst hier doch sicherlich die QNAP Oberfläche zum konfigurieren der VM. Innerhalb von RaspberryMatic selbst scheint der RF-USB ja nicht gefunden zu werden, sonst würde in der Hilfe an den besagten Stellen keine `n/a` auftauche wie im Screenshot zu sehen. Wie schon gesagt. SSH Login ins RaspberryMatic machen und mal mit `lsusb` schauen ob der Stick dort mit ID gelistet wird oder nicht. Oder als ReGaSkript sowas wie hier:

```c++
string stdout;
string stderr;
system.Exec("/usr/bin/lsusb", &stdout, &stderr);
WriteLine(stdout);
WriteLine(stderr);
```

Einfach unter "Skript testen" mal ausführen lassen und schauen was dabei rauskommt...

Und wenn der HmIP-RFUSB prinzipiell an RaspberryMatic richtig durchgeleitet wird muss sowas wie hier auftauchen:

```
Bus 002 Device 002: ID 1b1f:c020
````

mit der `ID 1b1f:c020` hinten.

**Comment by hardl1** [comment]
Danke schonmal für die Info.
Ich bin leider nicht zuhause um das prüfen zu können.

Außerdem muss ich mich einlesen, wie **man** per Login ins RaspberryMatic kommt oder ein ReGaSkript erstellt.

**Comment by hardl1** [comment]
Das scheint zu passen:

<img width="463" alt="Bildschirmfoto 2024-03-11 um 11 11 14" src="https://github.com/jens-maus/RaspberryMatic/assets/38688960/096fd29a-5c52-408d-b09d-6b7185cf1736">

**Reply by jens-maus** [reply]
Da ist das HmIP-RFUSB zu sehen, ja. Aber da wo du den `lsusb` befehl eingibst bist du nicht im Kontext von RaspberryMatic sondern das sieht nach der Kommandozeile deines QNAP-Systems aus. Ergo, SSH freischalten in der RaspberryMatic WebUI und dann dort einloggen oder eben in der RaspberryMatic WebUI dort unter "Skript testen" den Beispielskript von mir eingeben und ausführen lassen damit das auch **innerhalb** der RaspberryMatic ausgeführt wird.

**Comment by hardl1** [comment]
Sorry Jens, ich war zu doof um  "Skript testen" zu finden, obwohl ich schon damit Programme getestet habe.
Ich ärgere mich, dass ich Deine Zeit mit solchen Banalitäten beanspruchen musste.

Allerdings scheint der RF-USB erkannt zu werden.

<img width="304" alt="Bildschirmfoto 2024-03-11 um 15 16 35" src="https://github.com/jens-maus/RaspberryMatic/assets/38688960/34eb7bc2-bc85-4667-bba8-b2237b73970c">

**Comment by hardl1** [comment]
Der RF-USB wird erkannt, aber trotzdem kommt die Fehlermeldung:
<img width="931" alt="Bildschirmfoto 2024-03-13 um 08 23 59" src="https://github.com/jens-maus/RaspberryMatic/assets/38688960/eeb02a06-9f5b-4afc-9c65-7182e4006ff2">

**Reply by jens-maus** [reply]
Nein, er wird eben NICHT **erkannt**, sonst würde dieser ja unter der Hilfeseite auftauchen oder eben nutzbar sein.

Zeig mal den Inhalt der `/var/log/boot.log` Datei hier damit man sich ein Bild machen kann. Vmtl. wird es aber so sein wie @friedpa bereits geschrieben hat: Das QNAP System selbst "schnappt" sich den USB Stick weil es denkt es kann den nutzen und dann kann die VM diesen eben nicht mehr exklusiv nutzen.

P.S: Bitte hier nicht andauernd unter "Suggest an answer" neue Beiträge hinzufügen sondern bitte bei "Write a reply" deine Antwort eingeben, sonst wird das hier endlos lang und ich und andere verlieren die Lust dir zu antworten...

**Reply by hardl1** [reply]
Es gibt unter /var/log/ keine boot.log:
<img width="452" alt="Bildschirmfoto 2024-03-13 um 08 38 14" src="https://github.com/jens-maus/RaspberryMatic/assets/38688960/4d074a58-ca6b-420e-a4f9-e135cd7c2494">

**Reply by jens-maus** [reply]
Und wieder schaust du unter dem falschen System! QNAP Umgebung interessiert hier nicht. Schau innerhalb des RaspberryMatic Kontext nach einer `/var/log/boot.log`

**Reply by hardl1** [reply]
Sorry, aber ich finde die boot.log im Terminal nicht.
In der Raspi-Wartung finde ich nur eine "homematic-raspi-2024-03-13.log"

**Reply by hardl1** [reply]
Mein Problem ist beseitigt.

Im Fehlerprotokoll war ein fehlendes RedMatic Programm zu finden.
Das hat scheinbar den ordentlichen Bootvorgang verhindert.

<img width="1044" alt="Bildschirmfoto 2024-03-15 um 17 58 12" src="https://github.com/jens-maus/RaspberryMatic/assets/38688960/d109c988-c7fd-42de-b1e8-128feb6691b0">

Nach Installation von RedMatic läuft alles wieder.

**Reply by Baxxy13** [reply]
"Besser" wäre es gewesen nur das Überbleibsel der alten RedMatic Installation zu löschen
`rm -f /usr/local/etc/monit-redmatic.cfg`

Eigentlich sollte das RaspberryMatic selbst tun.

**Reply by jens-maus** [reply]
> "Besser" wäre es gewesen nur das Überbleibsel der alten RedMatic Installation zu löschen
> 
> `rm -f /usr/local/etc/monit-redmatic.cfg`
> 
> Eigentlich sollte das RaspberryMatic selbst tun.

Das löscht nur den Link wenn der ins Nirvana zeigt. Aber in dem Fall scheint die cfg in der Tat noch existiert zu haben, nur der Rest der RedMatic Installation halt irgendwie nicht. Dann löscht er den Link natürlich nicht.

**Reply by hardl1** [reply]
Es läuft auch noch, nachdem ich RedMatic gelöscht habe.

**Reply by jens-maus** [reply]
Das liegt vermutlich daran weil diesmal die  Deinstallationsroutinen von RedMatic komplett durchgelaufen sind, was vorher anscheinend nicht funktioniert hat…

**Reply by Baxxy13** [reply]
Ist schon echt unschön was dieses RedMatic - Überbleibsel für Auswirkungen haben kann.
Ich hatte mal bei jemandem per TeamViewer supportet dessen RM auch total instabil war.
Ursache war auch hier die `monit-redmatic.cfg`. 
Die Datei gelöscht, Monit neu gestartet und der Fall war erledigt. 😉
